### PR TITLE
add LogbackJsonLogViewer plugin

### PR DESCRIPTION
**Pull request by neno--**
- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [ ] My package doesn't add context menu entries. *
- [ ] My package doesn't add key bindings. **
- [ ] Any commands are available via the command palette.
- [ ] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [ ] If my package is a syntax it doesn't also add a color scheme. ***
- [ ] If my package is a syntax it is named after the language it supports (without suffixes like "syntax" or "highlighting").
- [ ] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

My package is a viewer of log files that are stored in JSON format. It is particularly useful for Java Logback style files. It can highlight and toggle various severities etc.

It adds a context menu item only if the active document is indeed something that can be viewed as JSON formatted log.
In that case, a new tab is opened with a formatted log. Keybindings are only active in this tab.

[https://github.com/neno--/LogbackJsonLogViewer](https://github.com/neno--/LogbackJsonLogViewer)

There are no packages like it in Package Control.

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

### CODE REVIEW

**Comment by braver** [comment]
Thanks for your submission, some feedback:
- You should create the command to open the settings file you mention in the readme, and create a main menu entry as well.
- Your processing in is_visible is potentially quite heavy. If you're trying to figure out if a file is JSON, why don't you simply read the file's scope (ie. source.json)?

**Comment by braver** [comment]
👋🏻

**Comment by neno--** [comment]
Hello @braver thanks for your feedback:

- I added the command to open and edit configs. It is available from Settings > Package Settings > LogbackJsonLogViewer menu.
- I am not processing is_visible anymore. It was an attempt to use heuristics to see if the file was a good candidate for processing. These files would mostly consist of JSON, but some components could add lines that are not JSON. Now users can choose whether they want to view any file with this plugin.

**Comment by braver** [comment]
Please look for some examples on how to use edit_settings to open settings in Split View. Having separate menu entries for user and default settings feels like something from the nineties by now 😉 

> I am not processing is_visible anymore. It was an attempt to use heuristics to see if the file was a good candidate for processing.

Alright. We do like seeing a setting to disable context menu entries. This gives users some control over what shows up in that menu, which tends to get crowded because it's prime real-estate that all packages want to be in. If your package is not something I use every day, I don't need it there. Instead, all commands should be available via the command palette anyway (you might need to add it there still) and that's also the first place people go to look for functionality.

**Comment by neno--** [comment]
I used the edit_settings command to open plugin settings. Also, added the command to the command palette.

**Comment by braver** [comment]
Alright, that's better 👍🏻 

- Please have a look at the indentation in your keymap files, they look a bit wonky.
- For main menu entries where you're nesting inside existing entries (e.g. "Package Settings") please only provide the id without the caption (you would actually be overwriting the caption otherwise).
- Preferably implement a setting to disable the context menu item, since it's now unconditionally visible in all views.

**Comment by braver** [comment]
👋🏻

**Comment by neno--** [comment]
Hello @braver thanks for your feedback. I fixed all three issues.

**Comment by braver** [comment]
👋🏻

**Comment by neno--** [comment]
Hi @braver tag v2.0.0 created: https://github.com/neno--/LogbackJsonLogViewer/releases/tag/v2.0.0

File: repository/l.json [hunk 1]
```diff
@@ -1559,6 +1559,18 @@
 				}
 			]
 		},
+		{
+			"name": "LogbackJsonLogViewer",
+			"author": ["neno--"],
```
Comment (braver) [inline repository/l.json:1564]: Please remove the “author” field, it’s not needed since it’s just your username.

**Review by braver — CHANGES_REQUESTED** [review]

File: repository/l.json (hunk 1 above)
Comment (braver) [reply repository/l.json:1564]: Sorry for not catching that earlier!

**Comment by neno--** [comment]
Hi, @braver I added my full name.

**Pull request merged**